software/base/src/main/java/org/apache/brooklyn/entity/machine/MachineInitTasks.java

```diff
@@ -105,19 +108,24 @@ protected void openIptablesImpl(Iterable<Integer> inboundPorts, SshMachineLocati
                 iptablesRules.add(IptablesCommands.saveIptablesRules());
             }
             List<String> batch = Lists.newArrayList();
+
+            ByteArrayOutputStream outStream = new ByteArrayOutputStream();
```

Comment:
Thanks @iyovcheva - this certainly fixes it.

My concern is though that it only fixes it for this one command. If someone calls another command (e.g. uses `STOP_IPTABLES`) then that command's stdout/stderr won't get logged. I don't want us to have to include this in every method that wants to execute an ssh command.

It therefore feels like a more holistic approach would be good.

However, I'm fine with us merging this part for now and us thinking more about it.